torch/nn/utils/convert_parameters.py

```diff
@@ -0,0 +1,42 @@
+import torch
+from torch.autograd import Variable
+
+
+def parameters_to_vector(parameters):
+    """Convert parameters to one vector
+
+    Arguments:
+        parameters (Iterable[Variable]): an iterator of Variables that are the
+            parameters of a model.
+
+    Returns:
+        The parameters represented by a single vector
+    """
+    vec = []
+    for param in parameters:
+        vec.append(param.view(-1))
```

Comment:
this will fail if the parameters are in multiple devices.
For example, will fail if:
- param1 is on CPU
- param2 is on GPU
Can happen if you put part of your model on CPU and other part on GPU

It will also fail if:
- param1 is on GPU0
- param2 is on GPU1

Can happen if you do model parallelism over GPUs.

So i think it's worth adding an additional check:
- errors out when it sees parameters are not on same devices
OR
- gets the first param, and records it's device / location
- then all subsequent parameters are moved onto that device

For now I think erroring out is safer.

Reply:
you can for example use:

```python
param_device = param.get_device() if param.is_cuda else -1
if param_device != prev_param_device:
    raise TypeError("Found two parameters on different devices, this is currently not supported")
```

Reply:
@soumith It is now updated.

Reply:
you updated it and removed it again, maybe some `git` mistake?

Reply:
@soumith I saw from current file changes, it is the updated version, I think ?

Reply:
Okay i was confused probably.
an additional bug (or you have to give error).

right now torch.cat(vec) will fail with a bad error message from `torch.cat` if `vec` has parameters from multiple devices (CPU/GPU, GPU1/GPU2 etc.). Can you add similar error checking as you did below.

Reply:
after you make this change, this PR is good to merge. Thank you.